### PR TITLE
Add assert for null duration on snackbar

### LINKED ISSUE #22738: SnackBar doesn't check if passed Duration is null -> build() of Scaffold breaks.
# Steps to Reproduce

1. Create a Scaffold.
2. Call showSnackBar() with a SnackBar with `null` as the parameter for `duration:`.

This will not only prevent the SnackBar from showing but prevent the whole Page/Scaffold from building.

Minimum example:
```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Body(),
      ),
    );
  }
}

class Body extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: FlatButton(onPressed: () => Scaffold.of(context).showSnackBar(SnackBar(content: Text("Duration is null"),duration: null,)),),
    );
  }
}
```
## Logs
```
I/flutter (20483): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (20483): The following NoSuchMethodError was thrown building Scaffold(dirty, state:
I/flutter (20483): ScaffoldState#aa9d4(tickers: tracking 2 tickers)):
I/flutter (20483): The getter 'inMilliseconds' was called on null.
I/flutter (20483): Receiver: null
I/flutter (20483): Tried calling: inMilliseconds
I/flutter (20483):
I/flutter (20483): When the exception was thrown, this was the stack:
I/flutter (20483): #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)
I/flutter (20483): #4      ScaffoldState.build (package:flutter/src/material/scaffold.dart:1510:32)
I/flutter (20483): #5      StatefulElement.build (package:flutter/src/widgets/framework.dart:3742:27)
I/flutter (20483): #6      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3654:15)
I/flutter (20483): #7      Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #8      StatefulElement.update (package:flutter/src/widgets/framework.dart:3811:5)
I/flutter (20483): #9      Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #10     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #11     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #12     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #13     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #14     StatelessElement.update (package:flutter/src/widgets/framework.dart:3714:5)
I/flutter (20483): #15     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #16     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #17     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #18     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #19     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #20     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #21     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #22     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #23     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #24     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #25     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #26     StatefulElement.update (package:flutter/src/widgets/framework.dart:3811:5)
I/flutter (20483): #27     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #28     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #29     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #30     StatelessElement.update (package:flutter/src/widgets/framework.dart:3714:5)
I/flutter (20483): #31     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #32     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #33     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #34     StatefulElement.update (package:flutter/src/widgets/framework.dart:3811:5)
I/flutter (20483): #35     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #36     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #37     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #39     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #40     ProxyElement.update (package:flutter/src/widgets/framework.dart:3921:5)
I/flutter (20483): #41     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #42     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #43     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #44     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #45     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #46     StatefulElement.update (package:flutter/src/widgets/framework.dart:3811:5)
I/flutter (20483): #47     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #48     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #49     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #50     StatelessElement.update (package:flutter/src/widgets/framework.dart:3714:5)
I/flutter (20483): #51     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #52     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4673:14)
I/flutter (20483): #53     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #54     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #55     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #56     ProxyElement.update (package:flutter/src/widgets/framework.dart:3921:5)
I/flutter (20483): #57     Element.updateChild (package:flutter/src/widgets/framework.dart:2711:15)
I/flutter (20483): #58     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3665:16)
I/flutter (20483): #59     Element.rebuild (package:flutter/src/widgets/framework.dart:3507:5)
I/flutter (20483): #60     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2255:33)
I/flutter (20483): #61     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:653:20)
I/flutter (20483): #62     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
I/flutter (20483): #63     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
I/flutter (20483): #64     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
I/flutter (20483): #65     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:751:7)
I/flutter (20483): #67     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
I/flutter (20483): #68     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
I/flutter (20483): #69     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
I/flutter (20483): (elided 4 frames from package dart:async)
I/flutter (20483): ═════════════════════════════════════════════════════════════════

```

In the `build()` method of the Scaffold this bit finally breaks the whole thing:
```dart
          _snackBarTimer = new Timer(snackBar.duration, () { //snackBar.duration is null
            [...]
          });
```

After debugging way too long and not realizing that a null value was passed to the SnackBar I can just say that the Exception wasn't sufficient as a developer to directly tell what happend. At least for me.

Also I presume it shouldn't break the whole Page/Scaffold😉

I guess one should firstly assert if the `duration` parameter is null in the `SnackBar` and also maybe make it clearer when calling Timer() with a null Duration that this is the error.
So maybe assert something there in addition?

### PR DESCRIPTION
**Pull request by jonahwilliams**
Though it has a default value, a user can still pass a null duration which will cause an exception in the scaffold.

Fixes https://github.com/flutter/flutter/issues/22738

### CODE REVIEW

File: packages/flutter/test/material/snack_bar_test.dart [hunk 1]
```diff
@@ -662,4 +662,32 @@ void main() {
     expect(find.text('test'), findsNothing);
   });
 
+  testWidgets('Snackbar asserts if passed a null duration', (WidgetTester tester) async {
+      const Key tapTarget = Key('tap-target');
```
Comment (jslavitz) [inline packages/flutter/test/material/snack_bar_test.dart]: <-- You know what you did wrong :)

Comment (jonahwilliams) [reply packages/flutter/test/material/snack_bar_test.dart]: Ru-oh!

Comment (jonahwilliams) [reply packages/flutter/test/material/snack_bar_test.dart]: Fixed

**Comment by jslavitz** [comment]
LGTM!

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 2]
```diff
@@ -155,6 +155,7 @@ class SnackBar extends StatelessWidget {
     this.duration = _kSnackBarDisplayDuration,
     this.animation,
   }) : assert(content != null),
+       assert(duration != null),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/snack_bar.dart]: The constructor's doc should up be updated too.

**Pull request merged**